### ISSUE TITLE
Crash with Cotton 1.0.0-rc.2

### ISSUE THREAD

**Issue by MarioAndWeegee3** [Juuxel/Adorn#21]
~~~~
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 7/13/19 4:33 PM
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors!
	at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.logErrors(EntrypointUtils.java:45)
	at net.fabricmc.loader.entrypoint.hooks.EntrypointClient.start(EntrypointClient.java:34)
	at net.minecraft.class_310.method_1503(class_310.java)
	at net.minecraft.class_310.method_1514(class_310.java:395)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	Suppressed: java.lang.NoClassDefFoundError: io.github.cottonmc.cotton.registry.CommonItems
		at juuxel.adorn.lib.ModItems.<clinit>(ModItems.kt:24)
		at juuxel.adorn.Adorn.onInitialize(Adorn.kt:16)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointClient$$Lambda$1568.00000000165329C0.accept(Unknown Source)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.lambda$logErrors$0(EntrypointUtils.java:38)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils$$Lambda$1569.00000000161ECDE0.accept(Unknown Source)
		at java.util.ArrayList.forEach(ArrayList.java:1257)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.logErrors(EntrypointUtils.java:36)
		... 11 more
	Caused by: java.lang.ClassNotFoundException: io.github.cottonmc.cotton.registry.CommonItems
		at java.net.URLClassLoader.findClass(URLClassLoader.java:591)
		at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:946)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:891)
		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:160)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		... 18 more
	Suppressed: java.lang.NoClassDefFoundError: io.github.cottonmc.cotton.Cotton
		at io.github.cottonmc.resources.GenericResourceType.registerBlock(GenericResourceType.java:92)
		at io.github.cottonmc.resources.GenericResourceType.registerBlocks(GenericResourceType.java:105)
		at io.github.cottonmc.resources.GenericResourceType.registerAll(GenericResourceType.java:99)
		at io.github.cottonmc.resources.CommonResources.initialize(CommonResources.java:67)
		at io.github.cottonmc.resources.CottonResources.onInitialize(CottonResources.java:17)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointClient$$Lambda$1568.00000000165329C0.accept(Unknown Source)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.lambda$logErrors$0(EntrypointUtils.java:38)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils$$Lambda$1569.00000000161ECDE0.accept(Unknown Source)
		at java.util.ArrayList.forEach(ArrayList.java:1257)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.logErrors(EntrypointUtils.java:36)
		... 11 more
	Caused by: java.lang.ClassNotFoundException: io.github.cottonmc.cotton.Cotton
		at java.net.URLClassLoader.findClass(URLClassLoader.java:591)
		at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:946)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:891)
		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:160)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		... 21 more
	Suppressed: java.lang.NoClassDefFoundError: io.github.cottonmc.cotton.registry.CommonItems
		at marioandweegee3.toolbuilder.common.tools.HeadMaterials.<clinit>(HeadMaterials.java:18)
		at marioandweegee3.toolbuilder.ToolBuilder.onInitialize(ToolBuilder.java:52)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointClient$$Lambda$1568.00000000165329C0.accept(Unknown Source)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.lambda$logErrors$0(EntrypointUtils.java:38)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils$$Lambda$1569.00000000161ECDE0.accept(Unknown Source)
		at java.util.ArrayList.forEach(ArrayList.java:1257)
		at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.logErrors(EntrypointUtils.java:36)
		... 11 more
	Caused by: java.lang.ClassNotFoundException: io.github.cottonmc.cotton.registry.CommonItems
		at java.net.URLClassLoader.findClass(URLClassLoader.java:591)
		at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:946)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:891)
		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:160)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:874)
		... 18 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at net.fabricmc.loader.entrypoint.hooks.EntrypointUtils.logErrors(EntrypointUtils.java:45)
	at net.fabricmc.loader.entrypoint.hooks.EntrypointClient.start(EntrypointClient.java:34)
	at net.minecraft.class_310.method_1503(class_310.java)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.class_310.method_1514(class_310.java:395)
	at net.minecraft.client.main.Main.main(Main.java:154)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)

-- System Details --
Details:
	Minecraft Version: 1.14.3
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_212, Eclipse OpenJ9
	Java VM Version: Eclipse OpenJ9 VM (JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20190417_339 (JIT enabled, AOT enabled)
OpenJ9   - bad1d4d06
OMR      - 4a4278e6
JCL      - 5590c4f818 based on jdk8u212-b03), Eclipse OpenJ9
	Memory: 38259952 bytes (36 MB) / 523632640 bytes (499 MB) up to 9663676416 bytes (9216 MB)
	JVM Flags: 11 total; -Xoptionsfile=C:\Program Files\AdoptOpenJDK\jdk-8.0.212.03-openj9\jre\bin\compressedrefs\options.default -Xlockword:mode=default,noLockword=java/lang/String,noLockword=java/util/MapEntry,noLockword=java/util/HashMap$Entry,noLockword=org/apache/harmony/luni/util/ModifiedMap$Entry,noLockword=java/util/Hashtable$Entry,noLockword=java/lang/invoke/MethodType,noLockword=java/lang/invoke/MethodHandle,noLockword=java/lang/invoke/CollectHandle,noLockword=java/lang/invoke/ConstructorHandle,noLockword=java/lang/invoke/ConvertHandle,noLockword=java/lang/invoke/ArgumentConversionHandle,noLockword=java/lang/invoke/AsTypeHandle,noLockword=java/lang/invoke/ExplicitCastHandle,noLockword=java/lang/invoke/FilterReturnHandle,noLockword=java/lang/invoke/DirectHandle,noLockword=java/lang/invoke/ReceiverBoundHandle,noLockword=java/lang/invoke/DynamicInvokerHandle,noLockword=java/lang/invoke/FieldHandle,noLockword=java/lang/invoke/FieldGetterHandle,noLockword=java/lang/invoke/FieldSetterHandle,noLockword=java/lang/invoke/StaticFieldGetterHandle,noLockword=java/lang/invoke/StaticFieldSetterHandle,noLockword=java/lang/invoke/IndirectHandle,noLockword=java/lang/invoke/InterfaceHandle,noLockword=java/lang/invoke/VirtualHandle,noLockword=java/lang/invoke/PrimitiveHandle,noLockword=java/lang/invoke/InvokeExactHandle,noLockword=java/lang/invoke/InvokeGenericHandle,noLockword=java/lang/invoke/VarargsCollectorHandle,noLockword=java/lang/invoke/ThunkTuple -Xjcl:jclse29 -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx9G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		accuratenetherportals: Accurate Nether Portals 1.0.0
		adorn: Adorn 1.1.2+1.14.3
		alwaysdroploot: Always Drop Loot 3.5.0+mc1.14.1
		animalnet: MrMarnic's AnimalNet 1.9 Fabric 1.14.2
		anvil_fix: AnvilFix 1.1.2
		appleskin: AppleSkin 1.0.6
		aquarius: Aquarius 1.3.1+1.14.1
		autoconfig1: Auto Config v1 1.1.2+mc1.14.1
		autofish: Autofish 0.8.4
		bamboomod: Bamboo Mod 0.2.0
		battle-towers: Battle Towers 1.0.3
		beachslimes: Beach Slimes 1.2.0+mc1.14.1
		bettersleeping: Better Sleeping 0.2.0
		biomeinfo: Biome Info v1.1
		blockus: Blockus 1.1.12
		blue_endless_jankson: jankson 1.1.2
		bouncylife: Bouncy Life 1.1.2+1.14.1
		cakechomps: Cake Chomps FABRIC-1.14.2-3.0
		carrots: carrots 1.4.1
		carvemelon: Carved Melons 1.2.0
		cloth: Cloth Events 0.3.1+build.23
		cloth-config: Cloth Config 0.2.4+build.17
		cloth-config2: Cloth Config v2 0.2.0
		clothesline-fabric: Clothesline 0.0.16
		clothesline_motor: Tumble Dryer 1.1
		composableautomation: Composable Automation 0.5.0+mc1.14.2
		cotton: Cotton 1.0.0-rc.2
		cotton-cauldron: Cotton Cauldron 1.0.0-rc.1
		cotton-client-commands: Cotton Client Commands 0.4.2+1.14.3-SNAPSHOT
		cotton-commons: Cotton Commons 1.0.0-rc.1
		cotton-config: Cotton Config 1.0.0-rc.1
		cotton-datapack: Cotton Datapack 1.0.0-rc.1
		cotton-energy: Cotton Energy 1.5.0+1.14.3
		cotton-logging: Cotton Logging 1.0.0-rc.1
		cotton-player-events: Cotton Player Events 1.0.0-rc.2
		cotton-resources: Cotton Resources 1.2.2+1.14.2
		couplings: Couplings 1.0.0
		craftingpad: Crafting Pad 1.0.3
		crawl: Crawl 0.5.6
		crochet: Crochet 1.0.4
		cursedchests: CursedChests 2.3.21
		darkelevators: Elevators 1.1.3
		diui: DiUI 1.0.2
		easyexcavate: Easy Excavate 1.5.1
		extra-pressure-plates: Extra Pressure Plates 1.1.2
		extrapieces: Extra Pieces 1.3.0
		fabriblocks: FabriBlocks 1.1.5
		fabric: Fabric API 0.3.0+build.192
		fabric-api-base: fabric-api-base 0.1.0+59147463
		fabric-biomes-v1: fabric-biomes-v1 0.1.0+896c7fbb
		fabric-commands-v0: fabric-commands-v0 0.1.1+25fd0c52
		fabric-containers-v0: fabric-containers-v0 0.1.2+25fd0c52
		fabric-content-registries-v0: fabric-content-registries-v0 0.1.1+05f11b75
		fabric-crash-report-info-v1: fabric-crash-report-info-v1 0.1.0+59147463
		fabric-events-interaction-v0: fabric-events-interaction-v0 0.1.0+59147463
		fabric-events-lifecycle-v0: fabric-events-lifecycle-v0 0.1.0+59147463
		fabric-furnaces: Fabric Furnaces 1.1.5
		fabric-item-groups-v0: fabric-item-groups-v0 0.1.0+1e69a0a5
		fabric-keybindings: fabric-keybindings 0.1.0+93af775a
		fabric-keybindings-v0: fabric-keybindings-v0 0.1.1+7dfef87c
		fabric-language-kotlin: Fabric Language Kotlin 1.3.40+build.1
		fabric-loot-tables-v1: fabric-loot-tables-v1 0.1.0+43028fa6
		fabric-mining-levels-v0: fabric-mining-levels-v0 0.1.0+59147463
		fabric-models-v0: fabric-models-v0 0.1.0+59147463
		fabric-networking: fabric-networking 0.1.0+93af775a
		fabric-networking-blockentity-v0: fabric-networking-blockentity-v0 0.1.1+25fd0c52
		fabric-networking-v0: fabric-networking-v0 0.1.2+200eb5c2
		fabric-object-builders-v0: fabric-object-builders-v0 0.1.1+39ff57de
		fabric-registry-sync-v0: fabric-registry-sync-v0 0.2.2+f0d56800
		fabric-renderer-api-v1: fabric-renderer-api-v1 0.1.1+099c1e8b
		fabric-renderer-indigo: fabric-renderer-indigo 0.1.9+3c859cf7
		fabric-rendering-data-attachment-v1: fabric-rendering-data-attachment-v1 0.1.0+02a46d5b
		fabric-rendering-fluids-v1: fabric-rendering-fluids-v1 0.1.0+dc4c57c2
		fabric-rendering-v0: fabric-rendering-v0 0.1.1+6dad974e
		fabric-resource-loader: fabric-resource-loader 0.1.0+93af775a
		fabric-resource-loader-v0: fabric-resource-loader-v0 0.1.1+59147463
		fabric-tag-extensions-v0: fabric-tag-extensions-v0 0.1.1+c189dc5c
		fabric-textures-v0: fabric-textures-v0 0.1.4+792c1c35
		fabriccontrolling: Fabric Controlling 1.0.1
		fabricloader: Fabric Loader 0.4.8+build.155
		fastflyblockbreaking: Fast Fly Block Breaking 1.1.0
		flesh-2-leather: Flesh 2 Leather 0.1.4
		fmagnet: Fabric Magnet ${version}
		foamfix: FoamFix 0.12.1
		gamestart: Game Start 1.0.1
		gbfabrictools: GBfabrictools 1.0.0
		giftit: Gift It! 1.0.2+1.14.2
		glassential: Glassential 1.14-1.0.0
		grid: Grid 1.14-fabric0.2.7-1.4.1
		harvest: Harvest 1.0.4
		healthoverlay: Health Overlay 1.2.8
		iamverysmart: I Am Very Smart 1.4.0+mc1.14.1
		infinityfix: Infinity Fix 1.0.4
		inventorypause: Inventory Pause 1.0.0
		ironbarrels: Iron Barrels 1.0.1
		jankson: Jankson 1.0.0+j1.1.2
		leadvillagers: Lead Villagers 1.1.0+mc1.14.1
		leaf-decay: Leaf Decay 1.0.3
		leathercows: Leather Cows 1.0.0
		libblockattributes: LibBlockAttributes 0.4.7
		libblockattributes_core: LibBlockAttributes (Core) 0.4.7
		libblockattributes_fluids: LibBlockAttributes (Fluids) 0.4.7
		libblockattributes_items: LibBlockAttributes (Items) 0.4.7
		libcd: LibCapableData 1.3.0+1.14.3
		libgui: LibGui 1.1.0
		lightoverlay: Light Overlay 2.0
		lil-tater: Lil Tater 1.0.2
		liteconversion: Lite Conversion 1.1.1
		mbackpacks: Mpcs' Backpacks 1.1.4
		miners_horizon: Miner's Horizon 1.5.0
		miniblocks: Example Mod 1.0.0
		mm: Manningham Mills 1.6
		modmenu: Mod Menu 1.6.2-92
		mousewheelie: Mouse Wheelie 1.2.3+1.14.3-pre2
		mycommands: MyCommands Fabric 1.14.2-06
		net_fabricmc_stitch: stitch 0.2.1.61
		noitemburn: No Item Burn 1.0.0
		nomoreglowingpots: No More Glowing Pots 1.0.0
		notenoughwands: Not Enough Wands 1.7.5.beta
		offlineskins: OfflineSkins v2a-fabric
		optifabric: Optifabric 0.4.0
		org_jetbrains_annotations: annotations 16.0.3
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.3.40
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.3.40
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.3.40
		org_jetbrains_kotlinx_kotlinx-coroutines-core: kotlinx-coroutines-core 1.2.2
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.2.2
		org_slf4j_slf4j-api: slf4j-api 1.7.26
		org_slf4j_slf4j-simple: slf4j-simple 1.7.26
		org_zeroturnaround_zt-zip: zt-zip 1.13
		persistentcake: Persistent Cake 0.1.1-fabric
		polyester: Polyester 0.3.1+1.14.3
		redstonebits: Redstone Bits 1.0.4+mc1.14.1
		redstonetweaks: Redstone Tweaks 1.0.2
		refinements: Refinements 1.1.0-24
		roughlyenoughitems: Roughly Enough Items 2.9.6+build.138
		rtree-3i-lite-fabric: rtree-3i-lite for Fabric 0.2.0
		shulkerboxtooltip: Shulker Box Tootip 1.3.0+1.14.2
		silk: Silk API 1.2.4-43
		silkyspawners: Silky Spawners 1.0.2
		simple_pipes: SimplePipeTest 0.1.8
		simplefireworks: Simple Fireworks 1.0.0
		simplezoom: Simple Zoom 1.0.0
		smoothbedrock: Smooth Bedrock (Fabric) 1.1.4
		sodden: Sodden 1.0.1.2
		spiceoffabric: Spice of Fabric 1.2.0-beta.5+1.14.2
		stockpile: Stockpile 1.0.6+1.14.1
		subnauticredstone: Subnautic Redstone 1.0.0
		superores: Super Ores 2.0.1
		tinyfuel: Tiny Fuel 0.0.1
		toolbuilder: Tool Builder 1.2.0
		torcherino: Torcherino 1.8.49
		towelette: Towelette 1.7.2
		trapexpansion: Trap Expansion 1.1.0
		trident_fixer: Trident Fixer 1.0.0
		tweed: Tweed API 2.1.0-beta.9
		vanilla-hammers: Vanilla Hammers 1.4.1
		vanilla-toolsets: Vanilla Toolsets 1.1.2
		vanilladeathchest: VanillaDeathChest ${version}
		variablepaxels: VariablePaxels 1.14 Pre-Release 5-0.0.1
		vivatech: Vivatech 1.0.0-alpha.9+1.14.2
		voxelmap: VoxelMap 1.9.10
	Launched Version: fabric-loader-0.4.8+build.155-1.14.3
	LWJGL: 3.2.2 build 10
	OpenGL: NO CONTEXT
	GL Caps: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Resource Packs: ~~ERROR~~ NullPointerException: null
	Current Language: ~~ERROR~~ NullPointerException: null
	CPU: <unknown>
~~~~

**Comment by Juuxel** [comment]
This is already fixed for the next release (in 7103e26). You can use Cotton 0.8.3-pre (which is included in Adorn) for now.

**Comment by SDUBZ** [comment]
when is next release coming to curse just wondering
